src/EFCore.MySql/Extensions/MySqlPropertyExtensions.cs

```diff
@@ -102,13 +102,12 @@ private static void CheckValueGenerationStrategy(IProperty property, MySqlValueG
         /// <returns> <c>true</c> if compatible. </returns>
         public static bool IsCompatibleIdentityColumn(IProperty property)
         {
-            var type = property.ClrType;
-
-            return (type.IsInteger()
-                        || type == typeof(decimal)
-                        || type == typeof(DateTime)
-                        || type == typeof(DateTimeOffset))
-                   && !HasConverter(property);
+            var valueConverter = GetConverter(property);
+            var type = (valueConverter?.ProviderClrType ?? property.ClrType).UnwrapNullableType();
+            return type.IsInteger()
+                   || type == typeof(decimal)
+                   || type == typeof(DateTime)
```

Comment:
Wow, AUTO_INCREMENT on date/time types?

Reply:
Not really, we are using (or misusing) it here as a way to allow users to define a `CURRENT_TIMESTAMP` as the `DEFAULT` for a column. This can currently be done in multiple ways, one of them is using `ValueGeneratedOnAdd()` on a `DateTime` column.

Reply:
This will make more sense here, when we update our value generation related code for 5.0 and rename `IsCompatibleIdentityColumn` to `IsCompatibleWithValueGeneration`, as Npgsql and MSSQL already did for 3.0.